trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/signing/SigningServiceType.java

```diff
@@ -13,16 +13,33 @@
  */
 package io.trino.aws.proxy.spi.signing;
 
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Set;
+
+import static io.trino.aws.proxy.spi.signing.SigningTrait.S3V4_SIGNER;
+import static io.trino.aws.proxy.spi.signing.SigningTrait.STREAM_CONTENT;
 import static java.util.Objects.requireNonNull;
 
-public record SigningServiceType(String serviceName)
+public record SigningServiceType(String serviceName, Set<SigningTrait> signingTraits)
```

Comment:
Would this make sense to be an Enum?

Reply:
The original idea with `SigningServiceType` was that it could be extended by third-parties outside of this project. `AccessType` has gotten in the way of that. I'd like to address that in the future. This is better as a `record` so that it can be extended.

Reply:
Ah I see not what you meant in your previous comments. 

`AccessType` is going to be annoying since annotations can't use records (yet?)

Reply:
No they can't. What I've done in the past is to do something like:

```java
public @interface ResourceSecurity
{
    Class<? extends AccessType> accessType();
}
```

Where `AccessType` becomes a record. The code then instantiates the record (asserting a no-arg constructor). It's not the prettiest thing but it works.

Reply:
Ha that is cool. Do you want to add that here?

Reply:
Sure - I'll add that. Or is it better as a separate PR?

Reply:
I think it makes sense as part of this one